bench/workgen/runner/microbenchmark_tiered_base.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+# This is the tiered microbenchmark base class that has the common code needed
+# for other workloads
+#
+
+import sys
+
+from runner import *
+from wiredtiger import *
+from workgen import *
+
+class microbenchmark_tiered:
+
+    context = Context()
+    tiered_storage_config = ",tiered_storage=(bucket=./bucket,bucket_prefix=pfx-,local_retention=0,name=dir_store)"
+    tiered_storage_config += ",extensions=(./ext/storage_sources/dir_store/libwiredtiger_dir_store.so=(early_load=true))"
+    conn_config = ",cache_size=2G,log=(enabled,file_max=10MB)"   # explicitly added
+    conn_config += tiered_storage_config
+    conn = context.wiredtiger_open("create," + conn_config)
+    session = conn.open_session(None)
+    wtperf_table_config = "key_format=S,value_format=S," +\
+        "exclusive=true,allocation_size=4kb," +\
+        "internal_page_max=64kb,leaf_page_max=4kb,split_pct=100,"
+    table_config = "type=file"
+    tables = []
+    table_count = 4
+    for i in range(0, table_count):
+        tname = "table:test" + str(i)
+        table = Table(tname)
+        session.create(tname,wtperf_table_config + table_config)
+        table.options.key_size = 20
+        table.options.value_size = 100
+        tables.append(table)
```

Comment:
This will work for this usage, but you should know that each of these variables is a "class" variable vs. an "instance" variable.  So if someone reuses this class and makes two microbenchmark_tiered() objects, they will both share the "context", "tables" variable etc.  That's not usually expected.

I'd move this to be instance variables.  You'll need a `__init__` function to do this.  Having an init function will also allow us to have parameters - which I think would be helpful to create an 'aws' or 'dir_store' version of a `microbenchmark_tiered`.  Like
```
    tiered =. microbenchmark_tiered(storage_source=`aws`)
```
More about instance variables here: https://pynative.com/python-instance-variables/

Reply:
Refactored the code.

Reply:
Thanks, much better!